src/watertap_contrib/seto/costing/solar/trough.py

```diff
@@ -0,0 +1,161 @@
+        doc="Cost per kWh (thermal) for the trough plant thermal storage",
+    )
+
+    blk.contingency_frac_direct_cost = pyo.Var(
+        initialize=0.07,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Fraction of direct costs for contingency",
+    )
+
+    blk.tax_frac_direct_cost = pyo.Var(
+        initialize=0.05,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Fraction of direct costs applicable for sales tax",
+    )
+
+    blk.sales_tax_rate = pyo.Var(
+        initialize=0.05,
+        units=pyo.units.dimensionless,
+        bounds=(0, 1),
+        doc="Sales tax rate as applied to applicable capital costs",
+    )
+
+    blk.fixed_operating_by_capacity = pyo.Var(
+        initialize=8,
+        units=costing.base_currency / (pyo.units.kW * costing.base_period),
+        bounds=(0, None),
+        doc="Fixed operating cost of trough plant per kW capacity",
+    )
+
+    blk.variable_operating_by_generation = pyo.Var(
+        initialize=0.001,
+        units=costing.base_currency / (pyo.units.kWh * costing.base_period),
+        bounds=(0, None),
+        doc="Variable operating cost of trough plant per kWh generated",
+    )
+
+    blk.electricity_rate = pyo.Var(
+        initialize=0.06,
+        units=costing.base_currency / pyo.units.kWh,
+        bounds=(0, None),
+        doc="Cost per kWh of electricity that is used by plant operations",
+    )
+
+
+@register_costing_parameter_block(
+    build_rule=build_trough_cost_param_block, parameter_block_name="trough"
+)
+def cost_trough(blk):
+
+    trough_params = blk.costing_package.trough
+    make_capital_cost_var(blk)
+    make_variable_operating_cost_var(blk)
+    make_fixed_operating_cost_var(blk)
+
+    blk.direct_cost = pyo.Var(
+        initialize=0,
+        units=blk.config.flowsheet_costing_block.base_currency,
+        bounds=(0, None),
+        doc="Direct cost of trough plant",
+    )
+
+    blk.system_capacity = pyo.Var(
+        initialize=0,
+        units=pyo.units.kW,
+        bounds=(0, None),
+        doc="Heat capacity of trough plant",
+    )
+
+    blk.hours_storage = pyo.Var(
+        initialize=6,
+        units=pyo.units.hour,
+        bounds=(0, None),
+        doc="Rated hours of thermal storage",
+    )
+
+    blk.annual_generation = pyo.Var(
+        initialize=0,
+        units=pyo.units.kWh,
+        bounds=(0, None),
+        doc="Annual heat generation of trough plant",
+    )
+
+    blk.annual_electrical_load = pyo.Var(
+        initialize=0,
+        units=pyo.units.kWh,
+        bounds=(0, None),
+        doc="Annual electricity use of trough plant",
+    )
+
+    blk.direct_cost_constraint = pyo.Constraint(
+        expr=blk.direct_cost
+        == (
+            blk.system_capacity
+            * (
+                trough_params.cost_per_capacity_capital
+                - trough_params.base_storage_hours
+                * trough_params.cost_per_storage_capital
```

Comment:
So do you subtract that amount of thermal storage that is already assumed to be aggregated within the system cost? Is that the idea? Just want to make sure I am understanding correctly.

Reply:
Correct, I am manually adjusting the storage cost that is aggregated within the direct cost as PySAM is not automatically adjusting that cost after changes to storage size.